packages/web-components/@sfx/search-box/src/search-box.ts

```diff
@@ -0,0 +1,241 @@
+import { LitElement, customElement, html, property } from 'lit-element';
+import { SEARCHBOX_EVENT } from './utils';
+
+@customElement('sfx-search-box')
+/**
+ * This entity is responsible for receiving user input and dispatching events
+ * based on input.
+ * The entity also listens for events, and updates data accordingly.
+ */
+export default class SearchBox extends LitElement {
+  /**
+   * Text used as placeholder in search box.
+   */
+  @property({ type: String, reflect: true })
+  placeholderText = 'Type your search';
+  /**
+   * Search term generated via user input.
+   */
+  @property({ type: String, reflect: true }) searchTerm = '';
+  /**
+   * Boolean to determine whether or not to display search button.
+   */
+  @property({ type: Boolean, reflect: true })
+  searchButton = false;
+  /**
+   * Boolean to determine whether or not to display clear button.
+   */
+  @property({ type: Boolean, reflect: true })
+  clearButton = false;
+
+  constructor() {
+    super();
+    this.updateText = this.updateText.bind(this);
+  }
+
+  /*
+   * To be removed with introduction of Base.
+   */
+  createRenderRoot() {
+    return this;
+  }
+
+  /**
+   * Adds event listeners.
+   */
+  connectedCallback() {
+    super.connectedCallback();
+    window.addEventListener(
+      SEARCHBOX_EVENT.AUTOCOMPLETE_HOVER,
+      this.updateText
+    );
+  }
+
+  /**
+   * Removes event listeners.
+   */
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    window.removeEventListener(
+      SEARCHBOX_EVENT.AUTOCOMPLETE_HOVER,
+      this.updateText
+    );
+  }
+
+  /**
+   * Dispatches a search request event with the `searchTerm` property or string passed in.
+   * Invoked in response to user interactions: `enter` key or click on search button.
+   */
+  emitSearchEvent(query?: String) {
+    const term = query && typeof query === 'string' ? query : this.searchTerm;
+    const searchboxRequestEvent = new CustomEvent(
+      SEARCHBOX_EVENT.SEARCH_REQUEST,
+      {
+        detail: term,
+        bubbles: true
+      }
+    );
+    window.dispatchEvent(searchboxRequestEvent);
+  }
+
+  /**
+   * Dispatches an autocomplete request event.
+   * Invoked in response to user input: typed search term is greater than 3 characters.
+   *
+   * @param letters A string of letters inputed into input box.
+   */
+  emitAutocompleteRequestEvent(letters: String) {
+    const autocompleteRequestEvent = new CustomEvent(
+      SEARCHBOX_EVENT.AUTOCOMPLETE_REQUEST,
+      { detail: letters, bubbles: true }
+    );
+    window.dispatchEvent(autocompleteRequestEvent);
+  }
+
+  /**
+   * Dispatches an event notifying that the input box has been cleared.
+   * Invoked in response to user input: clear button, remove all letters
+   * using keypresses.
+   */
+  emitSearchBoxClearedEvent() {
```

Comment:
I think this was a request from me. It may be possible to handle it with what you described, but one way or another, we need to capture when the input is specifically cleared so we can act on it by closing the SAYT box. For now, we can probably just use Johann's suggestion of firing the "change" event that includes the current content of the search box.

For the future, I think the question is: are we okay with two events firing when the user clears the search box? The input would now have zero characters (one event), and the Clear button would have been clicked (another event).